modules/core/src/main/java/org/apache/synapse/config/xml/CallMediatorFactory.java

```diff
@@ -91,8 +93,15 @@ public Mediator createSpecificMediator(OMElement elem, Properties properties) {
                 callMediator.setEndpoint(endpoint);
             }
         }
-
+        // create the blocking attribute and set it in the call mediator
+        OMAttribute blocking = elem.getAttribute(BLOCKING_Q);
+        if (blocking != null) {
+            callMediator.setBlocking(Boolean.parseBoolean(blocking.getAttributeValue()));
+        } else {
+            callMediator.setBlocking(false);
```

Comment:
CallOutMediator's blocking variable has a default value right (blocking=false)? Hence, why we need the else clause? That way you allow, CallOutMediator to decide the default value.